flocker/control/schema/types.yml

```diff
@@ -59,7 +59,17 @@ definitions:
         external:
           type: integer
     uniqueItems: true
-    
+
+  environment:
+    title: "Container environment variables"
+    description: "Mapping of label and value pairs for environment variables"
+    type: object
+    patternProperties:
+      "^.{1,4096}$":
+        type: string
+        maxLength: 4096
+    additionalProperties: false
```

Comment:
`maxProperties` - `metadata` in the types schema uses this if you want an example.

There's some maximum environment size supported by `exec(3)`.  Picking a limit at or below that is probably sensible.  I don't think the limit is actually defined by any standard, though (so it's a case of "whatever random value some random version of Linux picked", unfortunately).  On 3.13, seems to be in the ballpark of 2 MiB.  If we stick to 4096 for key and value limit, that's 256 items.  So let's go with that.